src/org/zaproxy/zap/extension/cspscanner/ContentSecurityPolicyScanner.java

```diff
@@ -177,17 +183,17 @@ private String getCSPNoticesString(ArrayList<Notice> notices) {
 		ArrayList<Notice> errorsList = Notice.getAllErrors(notices);
 		if (!errorsList.isEmpty()) {
 			returnSb.append(Constant.messages.getString(MESSAGE_PREFIX + "notices.errors")).append(NEWLINE);
-			for (int idx = 0; idx == errorsList.size() - 1; idx++) {
-				returnSb.append(errorsList.get(idx).show()).append(NEWLINE);
+			for (Notice notice : errorsList) {
+				returnSb.append(notice.message).append(NEWLINE);

```

Comment:
Was the `show()` dropped on purpose? The exact location of the error could be handy?